### DISCUSSION TITLE
Clickhouse queries

### DISCUSSION THREAD

**Discussion by hulakv** [burningalchemist/sql_exporter#542] [Q&A] [answered]
Hello. Trying execute query in clickhouse.

target format: clickhouse://user:password@hostname

collectorFiles:
  ch_metrics.collector.yml:
    collector_name: ch_metrics
    metrics:
      - metric_name: ch_query_duration
        type: gauge
        help: 'slow query duration count'
        values: [count]
        query: |
          SELECT COUNT(*) as count FROM system.query_log WHERE query_duration_ms > 25000

Sql-exporter getting an error:

`ts=2024-06-14T10:19:39.075Z caller=klog.go:134 level=error func=Errorf msg="Error gathering metrics: [from Gatherer #1] [job=ch_target,target=ch,collector=ch_metrics,query=ch_query_duration] code: 62, message: Syntax error: failed at position 80 ('VALUES'): VALUES. Expected one of: token, DoubleColon, OR, AND, IS NOT DISTINCT FROM, IS NULL, IS NOT NULL, BETWEEN, NOT BETWEEN, LIKE, ILIKE, NOT LIKE, NOT ILIKE, REGEXP, IN, NOT IN, GLOBAL IN, GLOBAL NOT IN, MOD, DIV, alias, AS, GROUP BY, WITH, HAVING, WINDOW, ORDER BY, LIMIT, OFFSET, FETCH, SETTINGS, UNION, EXCEPT, INTERSECT, INTO OUTFILE, FORMAT, end of query"`

This query successfully ran manually. Could you please help?

**Comment by burningalchemist** [comment]
Hi @hulakv, please check this discussion - https://github.com/burningalchemist/sql_exporter/discussions/520 and let me know if it didn't help.

Cheers!

**Comment by hulakv** [comment, marked as answer]
Thanks a lot!!
Added  **no_prepared_statement: true**  to collector metrics config and it's worked out. 
collector_name: ch_metrics
metrics:
      - metric_name: ch_query_duration
        no_prepared_statement: true
        type: gauge
